[kahsuann] iP

kahsuann opened the pull request:
# SusanLite
> *Just Do It. - Nike*

SusanLite will help you keep on top of 99 commitments! It's

- text-based
- easy to learn
- SUPER FAST to use

To start your productivity journey, simply:
1. download it from [here](https://github.com/kahsuann/ip/tree/master/src/main/java)
2. double-click it
3. add your takes
4. let Susan manage you tasks for you 😉 

and it is **FREE**!

Features:
- [X] Managing `tasks`
- [ ] Managing `deadlines` (coming soon)
- [ ] Reminders (coming soon)

[ TO NOTE ]
Here's a snippet of the Storage class to find the saved task list in your hard disk.
```
public class Storage {
    final String FILE_PATH = "./src/data/SusanToDoList.txt";

    public void load(TaskList tasks) throws IOException, SusanException {
        // Create data file
        File dataPath = new File("./src/data");
        if (!dataPath.exists()) {
            dataPath.mkdir();
        }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

kahsuann committed:
Initialised Susan's Greet and Exit

kahsuann committed:
Echo

kahsuann committed:
Add Tasks to List

kahsuann committed:
Task Mark as Done

kahsuann committed:
ToDos, Events, Deadlines

kahsuann committed:
Naive Implementation of Handling Errors

kahsuann committed:
Delete

kahsuann committed:
Automate Text UI Testing

kahsuann committed:
Merge and Complete Week 2 Tasks

kahsuann committed:
Save updated list to hard disk

kahsuann committed:
Clear Level-7 Bugs and Add Dates and Times

kahsuann committed:
Make the code more OOP with Ui, Storage, Parser, TaskList and Commands

kahsuann committed:
Organise into Packages

kahsuann committed:
Merge branch 'master' into add-gradle-support

kahsuann committed:
Use Gradle

kahsuann committed:
Add JUnit Tests

kahsuann committed:
Add JavaDoc comments

kahsuann committed:
Tweak the code to comply with a coding standard

kahsuann committed:
Find tasks with keyword

kahsuann committed:
Merge branch 'branch-A-CodingStandard'

kahsuann committed:
Merge branch 'branch-Level-9'

shizy commented on src/main/java/susan/command/Command.java:
```diff
@@ -0,0 +1,16 @@
+import java.io.IOException;
+
+public abstract class Command {
+    public abstract void execute(TaskList tasks, Ui ui, Storage storage) throws SusanException, IOException;
```
I like that you elected to use a verb to name the function!

shizy commented on src/main/java/susan/task/Task.java:
```diff
@@ -0,0 +1,34 @@
+        this.isDone = false;
+    }
+
+    public String getStatusIcon() {
```
It's great that you use the function name to explain the purpose of the function

shizy commented on src/main/java/susan/command/AddCommand.java:
```diff
@@ -0,0 +1,52 @@
+        }
+
+        Task task;
+        switch (taskType) {
```
Perhaps the switch case block could be formatted differently? (no indentation for case clauses)

shizy reviewed (commented):
Naming conventions are generally used well. Overall, LGTM! 👍

kahsuann committed:
Handle more exceptions

kahsuann committed:
Add data folder for Storage

kahsuann committed:
Update test cases

kahsuann committed:
Add more Javadoc comments

kahsuann committed:
Add minimal GUI

kahsuann committed:
Update GUI

arjun2598 commented on src/main/java/susan/command/AddCommand.java:
```diff
@@ -0,0 +1,71 @@
+package susan.command;
+
+import susan.task.*;
```
Perhaps the necessary files could be imported one by one explicitly?

arjun2598 commented on src/main/java/susan/task/ToDo.java:
```diff
@@ -0,0 +1,22 @@
+package susan.task;
+
+/**
+ * Represents a task that needs to be done without a stipulated deadline.
```
I like how every class is commented with detailed descriptions

arjun2598 commented on src/main/java/susan/ui/Parser.java:
```diff
@@ -0,0 +1,39 @@
+package susan.ui;
+
+import susan.command.*;
```
Perhaps the commands should be imported explicitly?

arjun2598 commented on src/main/java/susan/ui/Parser.java:
```diff
@@ -0,0 +1,39 @@
+ */
+public class Parser {
+
+    public static Command parse(String fullCommand) throws SusanException {
```
This parse method is very neat and clean 👍

arjun2598 commented on src/main/java/susan/ui/Storage.java:
```diff
@@ -0,0 +1,35 @@
+ * Handles saving tasks to file
+ */
+public class Storage {
+    final String FILE_PATH = "./src/data/SusanToDoList.txt";
```
Is the access modifier meant to be unspecified here?

arjun2598 commented on src/test/java/susan/command/AddCommandTest.java:
```diff
@@ -0,0 +1,55 @@
+    }
+
+    @Test
+    public void testWrongDateFormat() throws IOException {
```
I like that the tests are named appropriately 👍

arjun2598 reviewed (commented):
Overall, I think your code is done very well! It was easy to follow and the separation of code into many classes was nice to see. Just very minor coding standard violations that I noted.

kahsuann committed:
Use assertion

kahsuann committed:
Improve code quality

kahsuann committed:
Merge pull request #1 from kahsuann/branch-A-Assertions

Use assertion in Storage class

kahsuann committed:
Merge branch 'master' into branch-A-CodeQuality

# Conflicts:
#	src/main/java/susan/command/AddCommand.java

kahsuann committed:
Add remindme function

kahsuann committed:
Merge pull request #2 from kahsuann/branch-A-CodeQuality

Branch a code quality

kahsuann committed:
Merge pull request #2 from kahsuann/branch-A-CodeQuality

Branch a code quality

kahsuann committed:
Merge branch 'master' of https://github.com/kahsuann/ip

kahsuann committed:
Make FindCommand matching case-insensitive

Find command is case-sensitive.

A case-insensitive find is more user-friendly because
users cannot be expected to remember the exact case of the keywords.

Let's update the search algorithm to use case-insensitive matching.

kahsuann committed:
Make FindCommand matching case-insensitive

Find command is case-sensitive.

A case-insensitive find is more user-friendly because users cannot be expected to remember the exact case of the keywords.

Let's update the search algorithm to use case-insensitive matching.

kahsuann committed:
add a user guide

kahsuann committed:
Merge branch 'master' of https://github.com/kahsuann/ip

kahsuann committed:
update brief user guide and provide product screenshot

kahsuann committed:
Update brief user guide and provide product screenshot

kahsuann committed:
Update Javadoc comments

kahsuann committed:
Merge branch 'master' of https://github.com/kahsuann/ip

kahsuann committed:
Update JUnit testing

kahsuann committed:
Trim leading and trailing spaces from user input command

kahsuann committed:
Update Ui.png

kahsuann committed:
Save the task list properly after closing the program

kahsuann committed:
Close program properly upon exit command

kahsuann committed:
Update data filepath

kahsuann committed:
Update User Guide to reflect exit command

kahsuann committed:
Handle corrupted data file
